### PR TITLE
Propagate errors of ARO DNSMasqMachineConfig controller

### PR DESCRIPTION
**Pull request by SrinivasAtmakuri**
JIRA: https://issues.redhat.com/browse/ARO-3254

### Which issue this PR addresses:

Fixes

### CODE REVIEW

File: pkg/operator/controllers/dnsmasq/machineconfig_controller_test.go [hunk 1]
```diff
@@ -18,20 +20,28 @@ import (
...
-		objects    []client.Object
-		mocks      func(mdh *mock_dynamichelper.MockInterface)
-		request    ctrl.Request
-		wantErrMsg string
+		name            string
+		objects         []client.Object
+		mocks           func(mdh *mock_dynamichelper.MockInterface)
+		request         ctrl.Request
+		wantErrMsg      string
+		startConditions []operatorv1.OperatorCondition
```
Comment (tsatam) [inline pkg/operator/controllers/dnsmasq/machineconfig_controller_test.go:43]: Same suggestion as https://github.com/Azure/ARO-RP/pull/2951#discussion_r1254830989

**Review by tsatam — CHANGES_REQUESTED** [review]

File: pkg/operator/controllers/dnsmasq/machineconfig_controller_test.go (hunk 1 above)
Comment (SrinivasAtmakuri) [reply pkg/operator/controllers/dnsmasq/machineconfig_controller_test.go]: Changes made.

**Review by tsatam — APPROVED** [review]

File: pkg/operator/controllers/dnsmasq/machineconfig_controller_test.go [hunk 2]
```diff
@@ -126,10 +152,11 @@ func TestMachineConfigReconciler(t *testing.T) {
 				client,
 				dh,
 			)
-
+			ctx := context.Background()
 			_, err := r.Reconcile(context.Background(), tt.request)
```
Comment (jaitaiwan) [inline pkg/operator/controllers/dnsmasq/machineconfig_controller_test.go:156]: Nit: is creating another context for r.Reconcile necessary here. Can it just use the newly created ctx?

**Review by jaitaiwan — DISMISSED** [review]
LGTM. One nit but not really anything

**Review by yjst2012 — DISMISSED** [review]
LGTM, same nit as Daniel.

**Review by SudoBrendan — DISMISSED** [review]
LGTM, same nit.

File: pkg/operator/controllers/dnsmasq/machineconfig_controller_test.go (hunk 2 above)
Comment (SrinivasAtmakuri) [reply pkg/operator/controllers/dnsmasq/machineconfig_controller_test.go]: Made the change. Thanks!

**Review by jaitaiwan — APPROVED** [review]
LGTM

**Review by yjst2012 — APPROVED** [review]

**Review by petrkotas — APPROVED** [review]
LGTM

**Pull request merged**